babel/localedata.py

```diff
@@ -207,33 +209,32 @@ def resolve(self, data: Mapping[str | int | None, Any]) -> Mapping[str | int | N
         """
         base = data
         for key in self.keys:
-            data = data[key]
+            data = data[key]  # type: ignore
         if isinstance(data, Alias):
             data = data.resolve(base)
         elif isinstance(data, tuple):
             alias, others = data
             data = alias.resolve(base)
-        return data
-
+        return dict(data)
 
-class LocaleDataDict(abc.MutableMapping):
+class LocaleDataDict(MutableMapping[_Key, _Value]):
     """Dictionary wrapper that automatically resolves aliases to the actual
     values.
     """
 
-    def __init__(self, data: MutableMapping[str | int | None, Any], base: Mapping[str | int | None, Any] | None = None):
-        self._data = data
+    def __init__(self, data: MutableMapping[_Key, _Value], base: Mapping[_Key, _Value] | None = None):
+        self._data = dict(data)  # Storing as a dict allows copy() to work
         if base is None:
             base = data
-        self.base = base
+        self.base = dict(base)
```

Comment:
I have the feeling these `dict()` casts aren't a good idea... won't this imply that the `base` and `data` lose any specific dict type they are (e.g. `LocaleDataDict`) _and_ that there's an additional copy made..?

Should we maybe just do `cast(dict, ...)`?

Reply:
Internally they are probably dicts already.

An alternative for better type safety may be to make an ABC implementing Mapping with a copy method and use that as the annotation, or as a simpler approach, make the parameter annotations use dict